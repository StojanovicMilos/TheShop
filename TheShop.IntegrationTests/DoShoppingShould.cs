using System;
using System.Collections.Generic;
using System.Linq;
using ApprovalTests.Combinations;
using ApprovalTests.Reporters;
using TheShop.BL.Interfaces;
using TheShop.ConsoleApp;
using Xunit;

namespace TheShop.IntegrationTests
{
    public class DoShoppingShould
    {
        [Fact]
        public void OutputTextToConsoleForInitialUseCase()
        {
            //Setup
            var consoleOutput = new ConsoleOutput();
            string expectedOutput = "Debug: Trying to sell article with ID = 1"
                                    + Environment.NewLine + "Info: Article with ID = 1 is sold."
                                    + Environment.NewLine + "Found article with ID: 1"
                                    + Environment.NewLine + "Article with ID: 12 not found." + Environment.NewLine;
            SellRequest sellRequest = new SellRequest(1, 10, DateTime.Now, 458);
            List<int> getArticleIds = new List<int> {1, 12};
            Client client = Program.GetDefaultClient();

            //Exercise
            client.DoShopping(sellRequest, getArticleIds);

            //Verify
            Assert.Equal(expectedOutput, consoleOutput.GetOutput());

            //Teardown
            consoleOutput.Dispose();
        }

        [Fact]
        [UseReporter(typeof(DiffReporter))]
        public void OutputTextToConsoleForOtherUseCases()
        {
            //Arrange
            int[] orderAndSellArticleIds = Enumerable.Range(1, 2).ToArray();
            int[] buyerIds = Enumerable.Range(1, 5).ToArray();
            int[] maximumPrices = Enumerable.Range(457, 2).ToArray();
            TestList<TestList<int>> getArticleIds = new TestList<TestList<int>> {new TestList<int>(Enumerable.Range(1, 2))};

            //Act+Assert
            CombinationApprovals.VerifyAllCombinations(DoShoppingWithOutput, orderAndSellArticleIds, buyerIds, maximumPrices, getArticleIds);
        }

        private static string DoShoppingWithOutput(int orderAndSellArticleId, int buyerId, int maximumPrice, List<int> getArticleIds)
        {
            using (var consoleOutput = new ConsoleOutput())
            {
                Program.GetDefaultClient().DoShopping(new SellRequest(orderAndSellArticleId, buyerId, DateTime.Now, maximumPrice), getArticleIds);
                return consoleOutput.GetOutput();
            }
        }
    }
}
