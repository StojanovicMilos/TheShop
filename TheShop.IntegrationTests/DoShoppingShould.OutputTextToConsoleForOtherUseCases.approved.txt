[1,1,457,GetArticleIDs:1,2] => Could not order article
Article with ID: 1 not found.
Article with ID: 2 not found.

[1,1,458,GetArticleIDs:1,2] => Debug: Trying to sell article with ID = 1
Info: Article with ID = 1 is sold.
Found article with ID: 1
Article with ID: 2 not found.

[1,2,457,GetArticleIDs:1,2] => Could not order article
Article with ID: 1 not found.
Article with ID: 2 not found.

[1,2,458,GetArticleIDs:1,2] => Debug: Trying to sell article with ID = 1
Info: Article with ID = 1 is sold.
Found article with ID: 1
Article with ID: 2 not found.

[1,3,457,GetArticleIDs:1,2] => Could not order article
Article with ID: 1 not found.
Article with ID: 2 not found.

[1,3,458,GetArticleIDs:1,2] => Debug: Trying to sell article with ID = 1
Info: Article with ID = 1 is sold.
Found article with ID: 1
Article with ID: 2 not found.

[1,4,457,GetArticleIDs:1,2] => Could not order article
Article with ID: 1 not found.
Article with ID: 2 not found.

[1,4,458,GetArticleIDs:1,2] => Debug: Trying to sell article with ID = 1
Info: Article with ID = 1 is sold.
Found article with ID: 1
Article with ID: 2 not found.

[1,5,457,GetArticleIDs:1,2] => Could not order article
Article with ID: 1 not found.
Article with ID: 2 not found.

[1,5,458,GetArticleIDs:1,2] => Debug: Trying to sell article with ID = 1
Info: Article with ID = 1 is sold.
Found article with ID: 1
Article with ID: 2 not found.

[2,1,457,GetArticleIDs:1,2] => Could not order article
Article with ID: 1 not found.
Article with ID: 2 not found.

[2,1,458,GetArticleIDs:1,2] => Could not order article
Article with ID: 1 not found.
Article with ID: 2 not found.

[2,2,457,GetArticleIDs:1,2] => Could not order article
Article with ID: 1 not found.
Article with ID: 2 not found.

[2,2,458,GetArticleIDs:1,2] => Could not order article
Article with ID: 1 not found.
Article with ID: 2 not found.

[2,3,457,GetArticleIDs:1,2] => Could not order article
Article with ID: 1 not found.
Article with ID: 2 not found.

[2,3,458,GetArticleIDs:1,2] => Could not order article
Article with ID: 1 not found.
Article with ID: 2 not found.

[2,4,457,GetArticleIDs:1,2] => Could not order article
Article with ID: 1 not found.
Article with ID: 2 not found.

[2,4,458,GetArticleIDs:1,2] => Could not order article
Article with ID: 1 not found.
Article with ID: 2 not found.

[2,5,457,GetArticleIDs:1,2] => Could not order article
Article with ID: 1 not found.
Article with ID: 2 not found.

[2,5,458,GetArticleIDs:1,2] => Could not order article
Article with ID: 1 not found.
Article with ID: 2 not found.

